x/claim/types/claim_record.go

```diff
@@ -36,7 +36,7 @@ func (cr *ClaimRecord) ClaimAction(action Action) {
 
 // HasClaimedAction checks if the user has claimed a given action
 func (cr ClaimRecord) HasClaimedAction(action Action) bool {
-	return len(cr.ActionsCompleted) == len(Action_value)-1 && cr.ActionsCompleted[action-1]
+	return len(cr.ActionsCompleted) == len(Action_value)-1 || cr.ActionsCompleted[action-1]
```

Comment:
I don't understand this change? My idea was to check that the length of the slice is always 4 and that the user completed the action (index action-1) since the undefined action is not considered